scripts/build.sh

```diff
@@ -95,30 +105,41 @@ if [ "$JAVA_HOME" = "" ]; then
     echo "SET JAVA_HOME=$JAVA_HOME"
 fi
 
-cmake .
-make opensearchknn_faiss opensearchknn_nmslib
-
+# Build k-NN lib and plugin through gradle tasks
 cd $work_dir
-./gradlew assemble --no-daemon --refresh-dependencies -DskipTests=true -Dopensearch.version=$VERSION -Dbuild.snapshot=$SNAPSHOT -Dbuild.version_qualifier=$QUALIFIER
+# Gradle build is used here to replace gradle assemble due to build will also call cmake and make before generating jars
+./gradlew build --no-daemon --refresh-dependencies -x integTest -DskipTests=true -Dopensearch.version=$VERSION -Dbuild.snapshot=$SNAPSHOT -Dbuild.version_qualifier=$QUALIFIER
```

Comment:
Why are we excluding integration tests while building it ?

Reply:
Because due to this PR windows will fail to start cluster.
https://github.com/opensearch-project/OpenSearch/pull/4656